agent/agent_test.go

```diff
@@ -143,6 +143,7 @@ func TestRunProducts(t *testing.T) {
 	pCfg := product.Config{OS: "auto"}
 	p := make(map[string]*product.Product)
 	p["host"] = product.NewHost(pCfg)
+	// REVIEW(mkcp): Should we adapt this to call NewAgent? This seems like a symptom of poor boundaries between Agent and Products
```

Comment:
This would be in scope for agent cleanup -- the `agent_test.go` definitely has some signs of "greenfield development" and could be improved.

Reply:
I agree; it seems like it could be a call to NewAgent or else a test helper that could create an agent with a host product, since we use that in this test and the new one below.

Reply:
Added an issue for it [CORI-271](https://hashicorp.atlassian.net/browse/CORI-271)